packages/bundler-plugin-core/src/index.ts

```diff
@@ -389,6 +391,56 @@ const unplugin = createUnplugin<Options>((options, unpluginMetaContext) => {
         }
       },
     },
+    webpack(compiler) {
+      if (options._experiments?.debugIdUpload) {
+        compiler.hooks.compilation.tap("sentry-plugin", (compilation) => {
+          compilation.hooks.optimizeChunkAssets.tap("sentry-plugin", (chunks) => {
+            chunks.forEach((chunk) => {
+              const fileNames = chunk.files;
+              fileNames.forEach((fileName) => {
+                const source = compilation.assets[fileName];
+
+                if (!source) {
+                  logger.warn(
+                    "Unable to access compilation assets. If you see this warning, it is likely a bug in the Sentry webpack plugin. Feel free to open an issue at https://github.com/getsentry/sentry-javascript-bundler-plugins with reproduction steps."
+                  );
+                  return;
+                }
+
+                compilation.updateAsset(fileName, () => {
```

Comment:
do we need to cache here? So we only inject once?

Reply:
Added the cache inspired by https://github.com/webpack/webpack/pull/15454